### PR TITLE
Append bytes bigger than 0x79 properly

### LINKED ISSUE #1749: Can't correctly append bytes bigger than 0x79
When appending single bytes to `[]byte`, compiler adds extra `0x0` for all values starting from 0x80 up to 0xFF

```go
	{
		"append bytes bigger than 0x79",
		`package foo
		func Main() []byte {
			var z []byte
			z = append(z, 0x78, 0x79, 0x80, 0x81, 0xFF)
			return z
		}`,
		[]byte{0x78, 0x79, 0x80, 0x81, 0xFF},
	},

// Expected :[]byte{0x78, 0x79, 0x80, 0x81, 0xff}
// Actual   :[]byte{0x78, 0x79, 0x80, 0x0, 0x81, 0x0, 0xff, 0x0}
```

Appending with `z = append(z, []byte{0x78, 0x79, 0x80, 0x81, 0xFF}...)` solves the issue.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1749 .

I works, but I am currently thinking how we can handle byte varargs in a more general way, including user-defined functions.

### CODE REVIEW

**Comment by roman-khimov** [comment]
> NeoVM lacks opcode for array append

Like `APPEND`?

File: pkg/compiler/codegen.go [hunk 1]
```diff
@@ -1635,10 +1635,26 @@ func (c *codegen) convertBuiltin(expr *ast.CallExpr) {
 		} else {
 			emit.Instruction(c.prog.BinWriter, opcode.JMPIFNOT, []byte{2 + 2})
 			emit.Opcodes(c.prog.BinWriter, opcode.DROP, opcode.NEWARRAY0)
-			for _, e := range expr.Args[1:] {
-				emit.Opcodes(c.prog.BinWriter, opcode.DUP)
-				ast.Walk(c, e)
-				emit.Opcodes(c.prog.BinWriter, opcode.APPEND)
+			if expr.Ellipsis.IsValid() {
+				ast.Walk(c, expr.Args[1])                                       // append(x, y...) -> x, y on stack
+				emit.Opcodes(c.prog.BinWriter, opcode.DUP, opcode.REVERSEITEMS) // POPITEM pops last element
```
Comment (roman-khimov) [inline pkg/compiler/codegen.go:1640]: But this will change `y`.

Comment (fyrchik) [reply pkg/compiler/codegen.go:1640]: Fixed, inserted conversion to `Struct`, added test.

Comment (roman-khimov) [reply pkg/compiler/codegen.go:1641]: `REVERSEITEMS` is quite costly and now we also `CONVERT`, maybe it's easier with `PICKITEM`?

Comment (fyrchik) [reply pkg/compiler/codegen.go:1641]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**